classes.py

```diff
@@ -0,0 +1,544 @@
+
+        table = tables[0]
+        rows = table.find_all("tr")
+        for i in range(0, len(rows)):
+            cells = rows[i].find_all("td")
+            cell_data = [cell.get_text().strip() for cell in cells]
+            if cell_data[0] == "Name":
+                self.name = cell_data[1]
+            elif cell_data[0] == "Starting rank":
+                self.snr = cell_data[1]
+            elif cell_data[0] == "Ident-Number":
+                self.id = int(cell_data[1])
+                if not self.id:
+                    manual_entry_key = str(self.tournament.ord) + '.' + str(self.snr)
+                    if manual_entry_key in self.tournament.rating_cycle.manual_entries:
+                        self.id = self.tournament.rating_cycle.manual_entries[manual_entry_key]
+                    else:
+                        print()
+                        print('\tPlayer with unknown ID: %s' % self.name)
+                        self.id = int(input('\tPlease enter this player\'s ID: '))
+                        print()
+                        self.tournament.rating_cycle.manual_entries[manual_entry_key] = self.id
+
+        # Get Opponents and Results
+        table = tables[1]
+        header = table.select("tr")[0].find_all("td")
+        header_data = [h.get_text().strip() for h in header]
+        for cell_num, cell in enumerate(header_data):
+            if cell == 'SNo':
+                id_cell_num = cell_num
+            elif cell == 'Name':
+                name_cell_num = cell_num
+            elif cell == 'Res.':
+                result_cell_num = cell_num
+
+        rows = table.find_all("tr", recursive=False)
+        for i in range(1, len(rows)):
+            cells = rows[i].find_all("td", recursive=False)
+            id_opponent = cells[id_cell_num].get_text().strip()
+            name_opponent = cells[name_cell_num].get_text().strip()
+            result_opponent = cells[result_cell_num].get_text().strip()
+            if name_opponent not in ['not paired', 'bye']:
+                self.add_opponent(int(id_opponent), name_opponent, result_opponent)
+
+    def add_opponent(self, sno, name, result):
+        if result[-1] != "K":
+            if result[-1] == "½":
+                res = '0.5'
+            else:
+                res = result[-1]
+            self.opponents.append([sno, name, float(res)])
+
+    def keep_current_rating(self):
+        # self.new_rating = self.this_rating = self.last_rating
+        self.new_rating = self.last_rating
+        self.new_total_games = self.last_total_games
+        self.new_sum_oppon_ratings = self.last_sum_oppon_ratings
+        self.new_pts_against_oppon = self.last_pts_against_oppon
+
+    def calculate_new_rating(self, is_fexerj_tournament):
+
+        invalid_opponents = []
+        for k, tp_oppon in self.opponents.items():
+            if tp_oppon[0].is_unrated:
+                if self.is_unrated or (tp_oppon[0].new_rating is None) or tp_oppon[0].new_rating == 0:
```

Comment:
could you add parenthesis around this if condition and break lines around the ORs?

You can also replace the last 2 conditions by 

not tp_oppon[0].new_rating